File: ws/src/main/scala/de/thm/ii/fbs/model/Submission.scala
package de.thm.ii.fbs.model

import java.util.Date

/**
  * Users meta datate for a submission to a task.
  * @param submissionTime The time this submission was made
  * @param exitCode The exitcode of the check script, if 0, then the submission was a success
  * @param resultText The result text of the submission
  * @param checked If true then the submission was checked by the system
  * @param id The submission id
  */
case class Submission(submissionTime: Date, exitCode: Int, resultText: String, checked: Boolean, id: Int)

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/Submission.kt
package de.thm.ii.fbs.model.v2

import java.util.Date
import java.util.Optional

/**
 * Submission state
 *
 * @param submissionTime The submission time
 * @param done           True if submission was already checked
 * @param id             Submission id
 * @param results        The submission results
 */
data class Submission(
    val submissionTime: Date,
    val taskID: Int,
    val done: Boolean,
    val id: Int,
    val results: Array<CheckResult> = Array(),
    val userID: Optional<Int> = None,
    val isInBlockStorage: Boolean = false,
    val isHidden: Boolean = false
)
